### ISSUE TITLE
[FORMATTING] 'DESC' is not capitalised unlike other SQL keywords

### ISSUE THREAD

**Issue by avishj** [sql-formatter-org/sql-formatter#702]
**Input data**

Which SQL and options did you provide as input?

```sql
SELECT
    o.order_id
FROM
    public.orders o
    INNER JOIN public.order_details od ON od.order_id = o.order_id
    INNER JOIN public.product_detail pd ON pd.product_id = od.product_id
GROUP BY
    o.order_id
ORDER BY
    SUM(pd.weight) desc
LIMIT
    3;
```

**Expected Output**

```sql
SELECT
    o.order_id
FROM
    public.orders o
    INNER JOIN public.order_details od ON od.order_id = o.order_id
    INNER JOIN public.product_detail pd ON pd.product_id = od.product_id
GROUP BY
    o.order_id
ORDER BY
    SUM(pd.weight) DESC
LIMIT
    3;
```

**Actual Output**

```sql
SELECT
    o.order_id
FROM
    public.orders o
    INNER JOIN public.order_details od ON od.order_id = o.order_id
    INNER JOIN public.product_detail pd ON pd.product_id = od.product_id
GROUP BY
    o.order_id
ORDER BY
    SUM(pd.weight) desc
LIMIT
    3;
```

**Usage**

- How are you calling / using the library? -> Via the SQL Formatter VSCode Library by Rene
- What SQL language(s) does this apply to? -> Default SQL, Works as expected in PostgreSQL, TransactSQL.
- Which SQL Formatter version are you using? -> v4.0.0

**Comment by avishj** [comment]
The auto-detect / default SQL dialect should consider 'desc' as a keyword, since it is part of ANSI spec.

**Comment by nene** [comment]
Thanks for reporting.

There are two types of keywords in SQL: reserved and non-reserved ones. `DESC` was a reserved keyword since [SQL-99](https://ronsavage.github.io/SQL/sql-99.bnf.html), but it's no more since [SQL-2003](https://ronsavage.github.io/SQL/sql-2003-2.bnf.html) (it's just a plain keyword).

Anyway, the default "sql" dialect of SQL Formatter isn't actually intended to directly conform to the standard SQL. It's only there for cases where the user doesn't know which dialect of SQL he's dealing with. From that perspective, maybe the `ASC` and `DESC` keywords should get upper-cased. Currently, out of the 16 dialects supported by SQL Formatter, only 2 don't classify `ASC` and `DESC` as reserved keywords.

Therefore I agree that it should also be treated as reserved keyword by the default "sql" dialect.

**Comment by avishj** [comment]
Thanks, I was mistaken about DESC being a ANSI reserved keyword. Appreciate the quick changes!

**Comment by nene** [comment]
Fixed in 15.1.0 release.
